react/PaymentSummary.tsx

```diff
@@ -107,11 +112,39 @@ const PaymentSummary: React.FC = () => {
       break
     }
     default:
-      return lastStepsValid && isFreePurchase ? (
-        <Alert type="success">
-          <FormattedMessage id="store/checkout-payment.freePurchase" />
-        </Alert>
-      ) : null
+      if (lastStepsValid && isFreePurchase) {
+        return (
+          <Alert type="success">
+            <FormattedMessage id="store/checkout-payment.freePurchase" />
+          </Alert>
+        )
+      }
+
+      if (selectedPaymentSystem?.groupName.startsWith('custom')) {
```

Comment:
all custom payments will show the same summary if you use this logic.

Reply:
That's right. But I also don't have further information in the payment context to differentiate the custom payments 😢. Maybe we should get this info from the API somehow.

Reply:
Hey guys, I've worked with custom payments recently and I have a suggestion here.
There's a way to check for promissories exclusively, which is to check for the 3 digit number after the `custom` string.

I agree this shouldn't be the best approach but this is the way our payments-gateway checks for it. (I can't show it to you because the code is protected by PCI-DSS standards 😓). 

We only allow our stores to have at most 10 custom payments of each type (promissories, co-branded and privatelabels) and each have a fixed range of IDs. Promissories always varies from 201~210 so checking for it after `custom` would include only custom payments of promissories type.

[I've created a regex for it, hope it helps ;)](https://regexr.com/5rdmj)

Reply:
Nice!!!!

Reply:
Alright! Done in https://github.com/vtex-apps/checkout-payment/pull/29/commits/727c4d0528d1adb12aa4e36352626a1c4c8b24fb